xarray/plot/plot.py

```diff
@@ -661,6 +766,28 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
 
         _ensure_plottable(xval, yval)
 
+        # Replace pd.Intervals if contained in xval or yval.
+        if _valid_other_type(xval, [pd.Interval]):
+            if 'pcolormesh' == plotfunc.__name__:
+                xplt = _interval_to_bound_points(xval)
+                xlab_extra = ''
+            else:
+                xplt = _interval_to_mid_points(xval)
+                xlab_extra = '_center'
+        else:
+            xplt = xval
+            xlab_extra = ''
+        if _valid_other_type(yval, [pd.Interval]):
+            if 'pcolormesh' == plotfunc.__name__:
+                yplt = _interval_to_bound_points(yval)
+                ylab_extra = ''
+            else:
+                yplt = _interval_to_mid_points(yval)
+                ylab_extra = '_center'
+        else:
+            yplt = yval
+            ylab_extra = ''
```

Comment:
Can we put this logic in a helper function? This function is already getting pretty long :)